SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp

```diff
@@ -2047,11 +2047,44 @@ void ActorSystem::SendCreateEntityRequest(USpatialActorChannel& ActorChannel, ui
 	UE_LOG(LogActorSystem, Log, TEXT("Sending create entity request for %s with EntityId %lld, HasAuthority: %d"), *Actor->GetName(),
 		   ActorChannel.GetEntityId(), Actor->HasAuthority());
 
-	SpatialGDK::EntityFactory DataFactory(NetDriver, NetDriver->PackageMap, NetDriver->ClassInfoManager, NetDriver->RPCService.Get());
-	TArray<FWorkerComponentData> ComponentDatas = DataFactory.CreateEntityComponents(&ActorChannel, OutBytesWritten);
+	EntityFactory DataFactory(NetDriver, NetDriver->PackageMap, NetDriver->ClassInfoManager, NetDriver->RPCService.Get());
+	TArray<FWorkerComponentData> ComponentDatas;
+	TArray<FWorkerComponentUpdate> ComponentUpdates;
+
+	enum EEntityCreationType
+	{
+		CreateNewEntity,
+		PopulateSkeleton,
+	};
+
+	const EntityViewElement* ExistingEntity = NetDriver->Connection->GetCoordinator().GetView().Find(ActorChannel.GetEntityId());
+	const bool bIsSkeletonEntity =
+		ExistingEntity != nullptr
+		&& ExistingEntity->Components.ContainsByPredicate(ComponentIdEquality{ SpatialConstants::SKELETON_ENTITY_QUERY_TAG_COMPONENT_ID });
+	const EEntityCreationType EntityType = bIsSkeletonEntity ? PopulateSkeleton : CreateNewEntity;
+	if (EntityType == CreateNewEntity)
+	{
+		ComponentDatas = DataFactory.CreateEntityComponents(&ActorChannel, OutBytesWritten);
+	}
+	else if (EntityType == PopulateSkeleton)
+	{
+		if (ensure(ExistingEntity != nullptr))
```

Comment:
We've just checked this condition a couple of lines above. It doesn't seem like anything could have happened to reverse the result.